VideoLocker/src/main/java/org/edx/mobile/base/BaseFragmentActivity.java

```diff
@@ -531,37 +523,22 @@ protected void hideSoftKeypad() {
     }
 
     public boolean showErrorMessage(String header, String message) {
-        return showErrorMessage(header, message, true);
-    }
-
-    public boolean showErrorMessage(String header, String message, boolean isPersistent) {
-        LinearLayout error_layout = (LinearLayout) findViewById(R.id.error_layout);
-        if (error_layout == null) {
-            logger.warn("Error Layout not available, so couldn't show flying message");
-            return false;
-        }
-        TextView errorHeader = (TextView) findViewById(R.id.error_header);
-        TextView errorMessageView = (TextView) findViewById(R.id.error_message);
-        if (header == null || header.isEmpty()) {
-            errorHeader.setVisibility(View.GONE);
-        } else {
-            errorHeader.setVisibility(View.VISIBLE);
-            errorHeader.setText(header);
-        }
-        if (message != null) {
-            errorMessageView.setText(message);
-        }
-        ViewAnimationUtil.showMessageBar(error_layout, isPersistent);
+        hideErrorMessage();
+        alertDialog = new AlertDialog.Builder(this)
```

Comment:
Instead of `AlertDialog`, we should use `DialogFragment`. More discussion on this can be [found here] (https://github.com/edx/edx-app-android/pull/709#discussion_r66421643).

Speaking of which, you can use this class `AlertDialogFragment` that I created specifically for this reason, or `SimpleAlertDialog` that also accepts a bundle containing a `title` and `message`.